tsl/src/continuous_aggs/common.c

```diff
@@ -702,14 +702,28 @@ cagg_validate_query(const Query *query, const bool finalized, const char *cagg_s
 			foreach (l, query->jointree->fromlist)
 			{
 				Node *jtnode = (Node *) lfirst(l);
-				JoinExpr *join = NULL;
 				if (IsA(jtnode, JoinExpr))
 				{
-					join = castNode(JoinExpr, jtnode);
+					JoinExpr *join = castNode(JoinExpr, jtnode);
 					jointype = join->jointype;
 					op = (OpExpr *) join->quals;
-					rte = list_nth(query->rtable, ((RangeTblRef *) join->larg)->rtindex - 1);
-					rte_other = list_nth(query->rtable, ((RangeTblRef *) join->rarg)->rtindex - 1);
+
+					int lindex = 0, rindex = 0;
+
+					if (IsA(join->larg, RangeTblRef))
+						lindex = ((RangeTblRef *) join->larg)->rtindex;
+
+					if (IsA(join->larg, JoinExpr))
+						lindex = ((JoinExpr *) join->larg)->rtindex;
+
+					if (IsA(join->rarg, RangeTblRef))
+						rindex = ((RangeTblRef *) join->rarg)->rtindex;
+
+					if (IsA(join->rarg, JoinExpr))
+						lindex = ((JoinExpr *) join->rarg)->rtindex;
+
+					rte = list_nth(query->rtable, lindex - 1);
+					rte_other = list_nth(query->rtable, rindex - 1);
```

Comment:
At this point `rindex` can be zero if neither type matched above, which would lead to passing -1 to `list_nth`. You should probably deal with this in some manner to avoid crashes in release builds.

Reply:
It can't happen so added an assertion before.